#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("publish_status", bool_t, 0, "Publish study status", False)

trial_setup = gen.add_group("01 - Trial", type="tab")
task_setup = gen.add_group("02 - Task", type="tab")
user_setup = gen.add_group("03 - User", type="tab")

trial_setup.add("next_trial", bool_t, 0, "Switch to the next task from the list", False)
trial_setup.add("trial", str_t, 0, "Set current trial", "-1")

task_setup.add("next_task", bool_t, 0, "Switch to the next task from the list", False)
task_setup.add("task_id", str_t, 0, "Set current task id", "T1")
task_setup.add("task_stop_commands", bool_t, 0, "Switch to the next task from the list", False)

user_setup.add("next_user", bool_t, 0, "Switch to the next task from the list", False)
user_setup.add("user_id", str_t, 0, "Set current user id", "-1")

exit(gen.generate("classical_user_performance", "user_study_manager", "UserStudyManager"))

